src/byrow.jl

```diff
@@ -51,31 +51,27 @@ function byrow_helper(df, body)
     (e_body, e_newcols) = byrow_find_newcols(body, Any[])
     e_delimiters = Expr(:(=), :row, :( 1:_N ))
     e_forloop = Expr(:for, e_delimiters, byrow_replace(e_body))
-    return quote
-        _N = length($df[1])
-        _DF = @transform($df, $(e_newcols...))
-        @with _DF $e_forloop
-        _DF
-    end
+    :(_N = length($df[1]); _DF = @transform($df, $(e_newcols...)); @with _DF $e_forloop; _DF)
```

Comment:
P.S. this macro (as with all macros here) are completely escaped such that both _N and _DF will end up in user code. I was originally thinking this was a bug, because a user could get unexpected results when referencing these variables, but in fact, an aspiring user could use this strategically. Maybe it deserves documentation?